src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -137,12 +145,34 @@ public DatabaseDocumentActions(DocumentDatabase database, BuildVersionType build
 
             public void WriteDocument(Document document)
             {
-                _command.Add(document);
+                _command.Add(new DocumentItem
+                {
+                    Type = DocumentType.Document,
+                    Document = document,
+                });
+
+                HandleBatchOfDocumentsIfNecessary();
+            }
+
+            public void WriteAttachment(StreamSource.AttachmentStream attachment)
```

Comment:
Are we allowed to trigger a batch because of an attachment?
If we do, we might write it before the actual document was written, no?